__tests__/api/chat/channels.test.js

```diff
@@ -20,31 +46,88 @@ describe('GET /chat/channels/signed', () => {
       .set({Authorization: `Bearer valid-token`})
       .expect(200);
 
-    expect(response.body).toEqual(
-      expect.objectContaining({
-        code: 200,
-        data: expect.arrayContaining([
-          expect.objectContaining({
-            members: expect.arrayContaining([expect.any(Object)]),
-            messages: expect.any(String)
-          })
-        ]),
-        status: 'Success retrieve channels'
-      })
+    expect(response.body).toEqual(__getSignedChannelListBodyExpectation);
+    expectGetSignedChannelListFlow(user.user_id);
+
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      1,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 0)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      2,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 30)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      3,
+      ...__generateQueryChannelsParamsCall(user.user_id, 30, 60)
+    );
+    expect(StreamChat._instance.queryChannels).toHaveBeenNthCalledWith(
+      4,
+      ...__generateQueryChannelsParamsCall(user.user_id, 10, 90)
+    );
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test case seems to be making assumptions about the internal implementation of the `queryChannels` method. It's generally not a good practice to make assertions on how many times an internal function is called or with what arguments, as this makes your tests brittle and tightly coupled with the code implementation. Instead, focus on testing the observable behavior of the system (i.e., the returned result and any side effects). 



<!-- This is an auto-generated comment by OSS CodeRabbit -->